ci: zizmor: Address all issues

sprt opened the pull request:
This addresses all the Zizmor issues and reestablishes the test as required.

wainersm commented on .github/workflows/ci-on-push.yaml:
```diff
@@ -1,6 +1,6 @@
 name: Kata Containers CI
 on:
-  pull_request_target:
+  pull_request_target: # zizmor: ignore[dangerous-triggers]
```
This trigger is safe today but is still a weak point in our CI as if we by mistake remove some of the mitigations then it will be exposed. I'm saying that because I think we should leave an inline comment to always remember us of that risk. Disable the warning is okay.

sprt commented on .github/workflows/build-kata-static-tarball-amd64.yaml:
```diff
@@ -111,12 +111,15 @@ jobs:
           ARTEFACT_REGISTRY_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
           TARGET_BRANCH: ${{ inputs.target-branch }}
           RELEASE: ${{ inputs.stage == 'release' && 'yes' || 'no' }}
+          KBUILD_SIGN_PIN: ${{ secrets.KBUILD_SIGN_PIN }}
```
GH does not log secrets by design.

This is to satisfy Zizmor's template-injection audit https://docs.zizmor.sh/audits/#template-injection

sprt commented on .github/zizmor.yml:
```diff
@@ -0,0 +1,3 @@
+rules:
+  undocumented-permissions:
+    disable: true
```
I'm not a fan but I don't know the reasons for all 62 permissions that are assigned in these workflows so I couldn't fix all of them in this PR 🙂

stevenhorsman commented on .github/workflows/static-checks.yaml:
Do we need to call the command with a `bash -c` or something?

wainersm commented on .github/workflows/build-checks-preview-riscv64.yaml:
```diff
@@ -124,9 +124,11 @@ jobs:
-          cd ${{ matrix.component.path }}
-          ${{ matrix.command }}
+          cd ${COMPONENT_PATH}
+          ${COMMAND}
```
On the other hand, It doesn't seem likely to get a malicious code injected via 'matrix' context variable. maybe it is just a false-positive from zizmor.

stevenhorsman replied:
That's fair - if it's passing without needing to ignore, then lets push on with that

sprt commented:
Another fresh test run to fix the actionlint issues I accidentally introduced in the previous changes:
https://github.com/kata-containers/kata-containers/actions/runs/18289389944

sprt commented on .github/workflows/docs-url-alive-check.yaml:
```diff
@@ -15,8 +15,10 @@ jobs:
+        GITHUB_WORKSPACE: ${{ github.workspace }}
       run: |
-        echo "GOPATH=${{ github.workspace }}" >> "$GITHUB_ENV"
+        echo "GOPATH=${GITHUB_WORKSPACE}" >> "$GITHUB_ENV"
```
Oh I wasn't aware, thanks! Addressed this for all the affected vars I could find (GITHUB_REPOSITORY and GITHUB_WORKSPACE).

sprt commented:
Addressed @wainersm's comments and rebased on main.

One last round of tests:
https://github.com/kata-containers/kata-containers/actions/runs/18351903317
